app/kubemci/pkg/gcp/urlmap/urlmapsyncer_test.go

```diff
@@ -150,3 +159,218 @@ func TestDeleteURLMap(t *testing.T) {
 		t.Errorf("unexpected nil error, expected NotFound")
 	}
 }
+
+func TestListLoadBalancerStatuses(t *testing.T) {
+	lbName := "lb-name"
+	ipAddr := "1.2.3.4"
+	clusters := []string{"cluster1", "cluster2"}
+	svcName := "svcName"
+	svcPort := "svcPort"
+	ing := &v1beta1.Ingress{
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: svcName,
+				ServicePort: intstr.FromString(svcPort),
+			},
+		},
+	}
+	beMap := backendservice.BackendServicesMap{
+		svcName: &compute.BackendService{},
+	}
+	type ensureUM struct {
+		lbName   string
+		ipAddr   string
+		clusters []string
+		// True if the ensured forwarding rule should also have the right load balancer status.
+		// False by default.
+		hasStatus bool
+	}
+	testCases := []struct {
+		description  string
+		urlMaps      []ensureUM
+		expectedList map[string]status.LoadBalancerStatus
+	}{
+		{
+			"Should get an empty list when there are no url maps",
+			[]ensureUM{},
+			map[string]status.LoadBalancerStatus{},
+		},
+		{
+			"Load balancer should be listed as expected when url map exists",
+			[]ensureUM{
+				{
+					lbName,
+					ipAddr,
+					clusters,
+					true,
+				},
+			},
+			map[string]status.LoadBalancerStatus{
+				lbName: {
+					"",
+					lbName,
+					clusters,
+					ipAddr,
+				},
+			},
+		},
+		{
+			"Should ignore the load balancer without status",
+			[]ensureUM{
+				{
+					lbName + "1",
+					ipAddr,
+					clusters,
+					true,
+				},
+				{
+					lbName + "2",
+					ipAddr,
+					clusters,
+					false,
+				},
+			},
+			map[string]status.LoadBalancerStatus{
+				lbName + "1": {
+					"",
+					lbName + "1",
+					clusters,
+					ipAddr,
+				},
+			},
+		},
+	}
+	for i, c := range testCases {
+		glog.Infof("Test case: %d: %s", i, c.description)
+		frp := ingresslb.NewFakeLoadBalancers("" /*name*/, nil /*namer*/)
+		// Create all url maps.
+		for _, um := range c.urlMaps {
+			namer := utilsnamer.NewNamer("mci1", um.lbName)
+			ums := NewURLMapSyncer(namer, frp)
+			typedUMS := ums.(*URLMapSyncer)
+			if _, err := ums.EnsureURLMap(um.lbName, um.ipAddr, um.clusters, ing, beMap, false /*force*/); err != nil {
+				t.Errorf("expected no error in ensuring url map, actual: %v", err)
+			}
+			if !um.hasStatus {
+				if err := removeStatus(typedUMS); err != nil {
+					t.Fatalf("%s", err)
+				}
+			}
+		}
+		namer := utilsnamer.NewNamer("mci1", lbName)
+		ums := NewURLMapSyncer(namer, frp)
+		// List them to verify that they are as expected.
+		list, err := ums.ListLoadBalancerStatuses()
+		if err != nil {
+			t.Errorf("unexpected error listing load balancer statuses: %v", err)
+		}
+		if len(list) != len(c.expectedList) {
+			t.Errorf("unexpected list of load balancers, expected: %v, actual: %v", c.expectedList, list)
+		}
+		for _, rule := range list {
+			expectedRule, ok := c.expectedList[rule.LoadBalancerName]
+			if !ok {
+				t.Errorf("unexpected rule %v. expected rules: %v, actual: %v", rule, c.expectedList, list)
```

Comment:
`continue` to the next iteration instead of executing the rest of the loop body?

Reply:
done